In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

v=np.array([(10,1),(80,6),(37,3),(90,10),(31,2),(17,1),(50,4),(20,2),(73,4),(89,8)])
pureProfit=np.zeros(10)
for i in range(10):
    mult, hunt = v[i]
    pureProfit[i] = mult/hunt
pureProfit=np.round(pureProfit,2)
print("pureProfit: ",pureProfit)
thresh = np.zeros(10)
for i in range(10):
    mult, hunt = v[i]
    thresh[i] = (mult - 5*hunt)/500
print("thresh: ",thresh)

pureProfit:  [10.   13.33 12.33  9.   15.5  17.   12.5  10.   18.25 11.12]
thresh:  [0.01  0.1   0.044 0.08  0.042 0.024 0.06  0.02  0.106 0.098]


As we can see above, the expected profit is max for index 8 and 1 at 10.6% and 10% respectively. thus betting for a 2nd container might not be a wise strategy. This means we expect less than 10.6% (or 10%) of the players to choose this box and gain profit.

In [18]:
pureProfit=np.zeros(10)
for i in range(10):
    mult, hunt = v[i]
    pureProfit[i] = mult/hunt
pureProfit=np.round(pureProfit,2)
print("pureProfit: ",pureProfit)
share=np.array(pureProfit/np.sum(pureProfit))*100
share=np.round(share,2)
print("share: ",share)

pureProfit:  [10.   13.33 12.33  9.   15.5  17.   12.5  10.   18.25 11.12]
share:  [ 7.75 10.33  9.56  6.98 12.01 13.18  9.69  7.75 14.14  8.62]


If we neglect other players, we can see that the index 8 and index 5 give the max pure profit.
If we approximate the number of players going to any container using the pure profit mark, we can see that 8 and 5 have the highest attractiveness

In [19]:
maxmin_for1=np.zeros(10)
for i in range(10):
    mult, hunt = v[i]
    maxmin_for1[i] = mult*10000/(hunt+100)
maxmin_for1=np.round(maxmin_for1,1)
print("maxmin_for1: ",maxmin_for1)

maxmin_for1:  [ 990.1 7547.2 3592.2 8181.8 3039.2 1683.2 4807.7 1960.8 7019.2 8240.7]


In [20]:
maxmin_for2=float('-inf')
for i in range(10):
    mult, hunt = v[i]
    for j in range(10):
        if i != j:
            mult2, hunt2 = v[j]
            maxmin_for2 = max(mult*10000/(hunt+100)+mult2*10000/(hunt2+100)-50000, maxmin_for2)
print("maxmin_for2: ",maxmin_for2)

maxmin_for2:  -33577.44107744108


In a pessicimistic approach(considering everyone chooses the containers we choose), we can see that 2 containers will not give any profit. they will give a minimum loss of 33k

In [21]:
for k in range(10):
    shares = pureProfit**k
    shares = shares / np.sum(shares)
    maxmin_fori=float('-inf')
    l,n=-1,-1
    for i in range(10):
        mult, hunt = v[i]
        for j in range(10):
            if i != j:
                mult2, hunt2 = v[j]
                if mult*10000/(hunt+shares[i])+mult2*10000/(hunt2+shares[j])-50000 > maxmin_fori:
                    maxmin_fori = mult*10000/(hunt+shares[i])+mult2*10000/(hunt2+shares[j])-50000
                    l,n=i,j
    print("Exponent:", k, "Profit:", f"{maxmin_fori}", "Optimal expeditions:", l," ",n )

Exponent: 0 Profit: 282594.23503325944 Optimal expeditions: 5   8
Exponent: 1 Profit: 276476.73912702966 Optimal expeditions: 5   8
Exponent: 2 Profit: 270163.7125118715 Optimal expeditions: 5   8
Exponent: 3 Profit: 266311.0753600821 Optimal expeditions: 4   8
Exponent: 4 Profit: 263745.90780674433 Optimal expeditions: 4   8
Exponent: 5 Profit: 261668.33225030452 Optimal expeditions: 4   8
Exponent: 6 Profit: 260071.43833265238 Optimal expeditions: 4   8
Exponent: 7 Profit: 258893.74835951335 Optimal expeditions: 4   8
Exponent: 8 Profit: 258051.68875364214 Optimal expeditions: 4   8
Exponent: 9 Profit: 257461.70995565678 Optimal expeditions: 4   8


Last year the prediction for exponent ended up being the best. Also we can see that 8th index is the best always.